server/src/main/java/org/opensearch/cluster/routing/allocation/AllocationConstraints.java

```diff
@@ -8,58 +8,55 @@
 import org.opensearch.cluster.routing.allocation.allocator.BalancedShardsAllocator;
 import org.opensearch.cluster.routing.allocation.allocator.ShardsBalancer;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.function.Predicate;
 
+import static org.opensearch.cluster.routing.allocation.RebalanceConstraints.PREFER_PRIMARY_SHARD_BALANCE_NODE_BREACH_ID;
+import static org.opensearch.cluster.routing.allocation.RebalanceConstraints.PREFER_PRIMARY_SHARD_BALANCE_NODE_BREACH_WEIGHT;
+import static org.opensearch.cluster.routing.allocation.RebalanceConstraints.isPrimaryShardsPerIndexPerNodeBreached;
+
 /**
  * Allocation constraints specify conditions which, if breached, reduce the
- * priority of a node for receiving shard allocations.
+ * priority of a node for receiving unassigned shard allocations.
  *
  * @opensearch.internal
  */
 public class AllocationConstraints {
-    public final long CONSTRAINT_WEIGHT = 1000000L;
-    private List<Predicate<ConstraintParams>> constraintPredicates;
-
-    public AllocationConstraints() {
-        this.constraintPredicates = new ArrayList<>(1);
-        this.constraintPredicates.add(isIndexShardsPerNodeBreached());
-    }
 
-    class ConstraintParams {
-        private ShardsBalancer balancer;
-        private BalancedShardsAllocator.ModelNode node;
-        private String index;
-
-        ConstraintParams(ShardsBalancer balancer, BalancedShardsAllocator.ModelNode node, String index) {
-            this.balancer = balancer;
-            this.node = node;
-            this.index = index;
-        }
-    }
+    public final static long INDEX_SHARD_PER_NODE_BREACH_WEIGHT = 100000L;
 
     /**
-     * Evaluates configured allocation constraint predicates for given node - index
-     * combination; and returns a weight value based on the number of breached
-     * constraints.
-     *
-     * Constraint weight should be added to the weight calculated via weight
-     * function, to reduce priority of allocating on nodes with breached
-     * constraints.
      *
-     * This weight function is used only in case of unassigned shards to avoid overloading a newly added node.
-     * Weight calculation in other scenarios like shard movement and re-balancing remain unaffected by this function.
+     * This constraint is only applied for unassigned shards to avoid overloading a newly added node.
+     * Weight calculation in other scenarios like shard movement and re-balancing remain unaffected by this constraint.
      */
+    public final static String INDEX_SHARD_PER_NODE_BREACH_CONSTRAINT_ID = "index.shard.breach.constraint";
+    private Map<String, Constraint> constraintSet;
```

Comment:
nit, this is not a set: `private Map<String, Constraint> constraints;`

Reply:
Updated. 
Initially used the set that did not solve the purpose. Converted to map but use same naming variable.